FIAT/kong_mulder_veldhuizen.py

```diff
@@ -0,0 +1,165 @@
+#
+# contributions from Keith Roberts (University of Sao Paulo)
+#
+#
+# This file is part of FIAT (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+from FIAT import (
+    finite_element,
+    dual_set,
+    functional,
+    Bubble,
+    FacetBubble,
+    Lagrange,
+    NodalEnrichedElement,
+    RestrictedElement,
+)
+from FIAT.quadrature_schemes import create_quadrature  # noqa: F401
+
+
+def _get_topology(ref_el, degree):
+    """The topological association in a dictionary"""
+    T = ref_el.topology
+    sd = ref_el.get_spatial_dimension()
+    if degree == 1:  # works for any spatial dimension.
+        entity_ids = {0: dict((i, [i]) for i in range(len(T[0])))}
+        for d in range(1, sd + 1):
+            entity_ids[d] = dict((i, []) for i in range(len(T[d])))
+    elif degree == 2:
+        if sd == 2:
+            entity_ids = {
+                0: dict((i, [i]) for i in range(3)),
+                1: dict((i, [i + 3]) for i in range(3)),
+                2: {0: [6]},
+            }
+        elif sd == 3:
+            entity_ids = {
+                0: dict((i, [i]) for i in range(4)),
+                1: dict((i, [i + 4]) for i in range(6)),
+                2: dict((i, [i + 10]) for i in range(4)),
+                3: {0: [14]},
+            }
+    elif degree == 3:
+        if sd == 2:
+            etop = [[3, 4], [6, 5], [7, 8]]
+            entity_ids = {
+                0: dict((i, [i]) for i in range(3)),
+                1: dict((i, etop[i]) for i in range(3)),
+                2: {0: [9, 10, 11]},
+            }
+        elif sd == 3:
+            etop = [[4, 5], [7, 6], [8, 9], [11, 10], [12, 13], [14, 15]]
+            ftop = [[16, 17, 18], [19, 20, 21], [22, 23, 24], [25, 26, 27]]
+            entity_ids = {
+                0: dict((i, [i]) for i in range(4)),
+                1: dict((i, etop[i]) for i in range(6)),
+                2: dict((i, ftop[i]) for i in range(4)),
+                3: {0: [28, 29, 30, 31]},
+            }
+    elif degree == 4:
+        if sd == 2:
+            etop = [[6, 3, 7], [9, 4, 8], [10, 5, 11]]
+            entity_ids = {
+                0: dict((i, [i]) for i in range(3)),
+                1: dict((i, etop[i]) for i in range(3)),
+                2: {0: [i for i in range(12, 18)]},
+            }
+    elif degree == 5:
+        if sd == 2:
+            etop = [[9, 3, 4, 10], [12, 6, 5, 11], [13, 7, 8, 14]]
+            entity_ids = {
+                0: dict((i, [i]) for i in range(3)),
+                1: dict((i, etop[i]) for i in range(3)),
+                2: {0: [i for i in range(15, 30)]},
+            }
+    return entity_ids
+
+
+def bump(T, deg):
+    """Increase degree of polynomial along"""
+    sd = T.get_spatial_dimension()
+    if deg == 1:
+        return (0,)
+    if sd == 2:
+        if deg < 5:
+            return (1,)
+        elif deg == 5:
+            return (2,)
+    elif sd == 3:
+        return (1, 2)
+
```

Comment:
Catch not-implemented degrees in `sd==2` and `sd==3`.
Also catch not-implemented `sd`.

Reply:
Implemented the catches here.